src/subscriptions.ts

```diff
@@ -671,29 +667,38 @@ function hasBalanceChangeMatch(transactionBalanceChanges: BalanceChange[], filte
 /** Process an indexer transaction and return that transaction or any of it's inner transactions
  * that meet the indexer pre-filter requirements; patching up transaction ID and intra-round-offset on the way through.
  */
-function getFilteredIndexerTransactions(transaction: TransactionResult, filter: NamedTransactionFilter): SubscribedTransaction[] {
+function getFilteredIndexerTransactions(
+  transaction: algosdk.indexerModels.Transaction,
+  filter: NamedTransactionFilter,
+): SubscribedTransaction[] {
   let parentOffset = 0
   const getParentOffset = () => parentOffset++
 
   const transactions = [
-    { ...transaction, filtersMatched: [filter.name] } as SubscribedTransaction,
+    new SubscribedTransaction({ ...transaction, filtersMatched: [filter.name] }),
     ...getIndexerInnerTransactions(transaction, transaction, getParentOffset),
-  ]
+  ] satisfies SubscribedTransaction[]
   return transactions.filter(indexerPreFilterInMemory(filter.filter))
 }
 
 /** Return a transaction and its inner transactions as an array of `SubscribedTransaction` objects. */
-function getIndexerInnerTransactions(root: TransactionResult, parent: TransactionResult, offset: () => number): SubscribedTransaction[] {
-  return (parent['inner-txns'] ?? []).flatMap((t) => {
+function getIndexerInnerTransactions(
+  root: algosdk.indexerModels.Transaction,
+  parent: algosdk.indexerModels.Transaction,
+  offset: () => number,
+): SubscribedTransaction[] {
+  return (parent.innerTxns ?? []).flatMap((t) => {
     const parentOffset = offset()
+    const innerTxns = getIndexerInnerTransactions(root, t, offset)
     return [
-      {
+      new SubscribedTransaction({
         ...t,
         parentTransactionId: root.id,
```

Comment:
This is existing, but it's interesting this is called "parent..." but it uses root?